mne/io/persyst/persyst.py

```diff
@@ -0,0 +1,443 @@
+                      f'for {n_chs} channels.')
+
+        raw_extras = {
+            'datatype': datatype,
+            'n_chs': n_chs,
+            'n_samples': n_samples
+        }
+        # create Raw object
+        super(RawPersyst, self).__init__(
+            info, preload, filenames=[dat_fpath],
+            last_samps=[n_samples - 1],
+            raw_extras=[raw_extras], verbose=verbose)
+
+        # set annotations based on the comments read in
+        num_comments = len(comments_dict.keys())
+        onset = np.zeros(num_comments, float)
+        duration = np.zeros(num_comments, float)
+        description = [''] * num_comments
+        for t_idx, (_description, (_onset, _duration)) in \
+                enumerate(comments_dict.items()):
+            # extract the onset, duration, description to
+            # create an Annotations object
+            onset[t_idx] = _onset
+            duration[t_idx] = _duration
+            description[t_idx] = _description
+        annot = Annotations(onset, duration, description)
+        self.set_annotations(annot)
+
+        # elapsed time (in min)
+        elapsed = (time.time() - start_time) / 60
+        logger.info('Read in file in %s minutes.' % elapsed)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a segment of data from a file.
+
+        The Persyst software  records raw data in either 16 or 32 bit
+        binary files. In addition, it stores the calibration to convert
+        data to uV in the lay file.
+        """
+        datatype = self._raw_extras[fi]['datatype']
+        n_chs = self._raw_extras[fi]['n_chs']
+        dat_fname = self._filenames[fi]
+
+        # compute samples count based on start and stop
+        time_length_samps = stop - start
+
+        # read data from .dat file into array of correct size, then calibrate
+        # records = recnum rows x inf columns
+        if time_length_samps == -1:
+            count = -1  # elements of size precision to read
+        else:
+            count = time_length_samps * n_chs
+
+        # seek the dat file
+        # dat file
+        with open(dat_fname, 'rb') as dat_file_ID:
+            # allow offset to occur
+            if int(datatype) == 7:
+                precision = np.int32
+                dat_file_ID.seek(n_chs * 4 * start, 1)
+            else:
+                precision = np.int16
+                dat_file_ID.seek(n_chs * 2 * start, 1)
+
+            # read in the actual record starting at possibly offset
+            record = np.fromfile(dat_file_ID, dtype=precision,
+                                 count=count)
+
+        # chs * rows
+        record = np.reshape(record, (n_chs, -1), 'F')
+        # calibrate to convert to uV
+        data[...] = record[idx, ...] * cals
+        # cast as float32; more than enough precision
+        # then multiply to get V
+        data = data.astype(np.float32) / 10. ** 6
```

Comment:
`mult` is unused, so this code is going to be broken when projectors are used. You probably want to let `_mult_cal_one` take care of this stuff.

Also if you set `info['chs'][k]['cal']` appropriately above, you don't need to do the `1e-6` mult (here done as a div by `10 ** 6`) yourself. Setting it to 1e6 or 1e-6 (can't remember which) should have an equivalent effect.

Reply:
Okay I see, I'll adjust to multiply the info cal accordingly.

Re `mult_cal_one` sorry what is this? I've never worked w/ projectors :p

Reply:
Basically instead of multiplying by `cals` and sub-indexing yourself you should let `_mult_cal_one` do it. See how it's used in other readers by `git grep _mult_cal_one`.